internal/node/otel.go

```diff
@@ -470,64 +470,63 @@ func (dn *Node) ReceiveTrace(
 	return otel.GetTextMapPropagator().Extract(ctx, carrier)
 }
 
+// getSpan retrieves the current otel span from the context.
+func getSpan(ctx context.Context) trace.Span {
+	return trace.SpanFromContext(ctx)
+}
+
 // AddSpan adds a new otel span.  If the otel client is nil, no-ops.
-// Attrs can be added to the span with addSpanAttrs.  This span will
-// continue to be used for that purpose until replaced with another
-// span, which will appear in a separate context (and thus a separate,
-// node).
+// This span will continue to be used for that purpose until replaced
+// with another span, which will appear in a separate context (and thus
+// a separate, node).
 func (dn *Node) AddSpan(
 	ctx context.Context,
 	name string,
+	kvs map[string]any,
 	opts ...trace.SpanStartOption,
-) (context.Context, *Node) {
+) context.Context {
 	if dn == nil {
-		return ctx, dn
-	}
-
-	if dn.OTEL == nil {
-		// Create a new node so that it can have its properties set separately.
-		spawn := dn.SpawnDescendant()
-		spawn.ID = name
-
-		return ctx, spawn
+		return ctx
 	}
 
-	ctx, span := dn.OTEL.Tracer.Start(ctx, name, opts...)
-
 	spawn := dn.SpawnDescendant()
```

Comment:
it may be worth adding a comment somewhere in the function stating that a new clues node is also created. I'm assuming it's made so that the trace names in the clues nodes correspond to the names of the spans?